### ISSUE TITLE
`question_mark` false negative with and_then

### ISSUE THREAD

**Issue by xtqqczze** [rust-lang/rust-clippy#16893]
### Summary

.

### Lint Name

`question_mark`

### Reproducer

I tried this code:

```rust
make_fifo(to).and_then(|_| fs::remove_file(from))
```

https://github.com/uutils/coreutils/commit/2df9af64783e04aba34812a51d35b9d2e73879f3

I expected to see this happen:

Warning, with a suggestion

```rust
make_fifo(to)?;
fs::remove_file(from)
```

Instead, this happened:

No warning

### Version

```text
nightly
```

**Comment by QEEK-AI** [comment]
**QEEK.AI Agent:** 

"Thanks for the report! This is indeed a valid false negative , the `question_mark` lint should catch this pattern".

**The pattern:**

```rust
// Current (should lint)
make_fifo(to).and_then(|_| fs::remove_file(from))

// Suggested (more idiomatic)
make_fifo(to)?;
fs::remove_file(from)
```

**Why this makes sense:**

When `and_then` is used with a closure that ignores its argument (`|_|`) and simply returns another `Result`, it's semantically equivalent to:
1. Propagating the error with `?`
2. Then executing the next operation

This is a cleaner, more direct way to write sequential fallible operations.

**Edge cases to consider:**

The lint should probably only trigger when:
- The closure ignores its argument (or binds it with `_`)
- The closure body is a simple expression (not complex logic)
- The return type matches (both are `Result<_, _>`)

**Complex cases to avoid:**

```rust
// Should NOT lint uses the value
x.and_then(|v| do_something(v))

// Should NOT lint — complex closure body  
x.and_then(|_| {
    let y = compute();
    do_work(y)
})
```

This seems like a straightforward enhancement to the `question_mark` lint. The relevant logic would be in `clippy_lints/src/question_mark.rs` where it already handles similar patterns like `map_err` → `?` transformations.

Since @profetia has self-assigned, looking forward to seeing the implementation! Let me know if you need any help with the pattern matching logic.

**Comment by profetia** [comment]
I think `question_mark` should only lint if the `.and_then()` is in the function's return position.
